src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -157,7 +162,14 @@ public void Enqueue(TElement element, TPriority priority)
 
             // Restore the heap order
             int lastNodeIndex = GetLastNodeIndex();
-            MoveUp((element, priority), lastNodeIndex);
+            if (_comparer == null)
```

Comment:
This approach attempts to minimize the number of times we null check `_comparer`, but perhaps things would be simpler if we moved everything behind one `MoveUp`/`MoveDown` method.